core-v1/src/main/scala/algebra/Expr.scala

```diff
@@ -473,6 +472,59 @@ object Expr {
+    *       only way to construct this node would be to chain a previous operation with [[AndThen]]. This works
+    *       fine, however, it means that you can't attach a debugger to this [[Select]] node directly, and since
+    *       most of the complication of this expression is how the wrapper type is threaded through the output
+    *       of the applied lens, it makes sense to complicate this type to make it easier to debug custom
+    *       wrapper type logic.
+    * @note The type parameters [[A]] and [[B]] are invariant because of the [[wrapSelected]] function.
+    *       While invariant type parameters can be a hassle to work with from the interface user's perspective,
+    *       typically the caller only cares about the initial input type [[I]] or the final output type [[O]],
+    *       which both utilize variance.
+    *
+    *       The other option would have been to add an extra type parameter and rely on compiler provided
+    *       evidence that the types are compatible when the expression is constructed, but carrying around that
+    *       extra type parameter and compiler evidence is probably not worth its weight in code.
+    *
+    *       In any case where one would want to abstract over <i>specifically</i> the [[Select]] node, they
+    *       would have to accept a type parameter for the lens input type, because they cannot just operate
+    *       on the types [[Any]] / [[Nothing]]. This is most likely already the case, as those types are not
+    *       very useful, and if you don't care about the type, you can use the `_` wildcard.
+    *
+    *       It is a bit easier to add variance to a public interface than to take it away, so I opted to
+    *       make the input type parameters [[A]] and [[B]] invariant, rather than split them into two types each.
+    * @param inputExpr  the expression to run before applying the [[lens]] and [[wrapSelected]] on the result
+    * @param wrapSelected a function that defines how to wrap the result of applying the [[lens]].
+    *                   If it is a higher-kinded Functor, wrap every element, otherwise wrap the single value directly.
+    * @param lens       a lens from the input type to some selected field
+    * @tparam W the wrapper type (or effect) over which the select operation will operate.
+    *           Additionally, the resulting value will either be directly wrapped OR
+    *           if the selected type is a higher-kinded Functor, it will map over the elements
+    *           to wrap each individual element.
+    * @tparam A the output of the input expression and the input to the lens
+    * @tparam B the output of the lens
+    * @tparam O the final computed output type with the wrapper type applied at the appropriate level
+    *           (computed using the [[SelectOutputType]] implicit)
+    */
+  final case class Select[-I, W[+_] : Extract, A, B, +O : OP, OP[_]](
```

Comment:
In my latest PR, I actually need to remove the `W` as it turns out to be an implementation detail that the DSL can provide, and it gets in the way of a later feature. So I'm going to try and clean that up now.